### PR TITLE
nixos/install-tools: Add manpages to packages instead of seperating them

### LINKED ISSUE #244450: Documentation: `nixos-rebuild` manpages not included on non-nixos
## Problem

I'm using `nixos-rebuild` on non-nixos with the `--target-host` flag as a deployment tool. I have it in a devShell, like so:

```nix
devShells.default = pkgs.mkShell {
  buildInputs = [
    pkgs.nixos-rebuild
  ];
};
```
However, both `man nixos-rebuild` and `nixos-rebuild --help`, give
```
No manual entry for nixos-rebuild
```

## Proposal

I propose that the relevant manpage be included in the package.

## Checklist

- [x] checked [latest Nixpkgs manual] \([source][nixpkgs-source]) and [latest NixOS manual] \([source][nixos-source])
- [x] checked [open documentation issues] for possible duplicates
- [x] checked [open documentation pull requests] for possible solutions

[latest Nixpkgs manual]: https://nixos.org/manual/nixpkgs/unstable/
[latest NixOS manual]: https://nixos.org/manual/nixos/unstable/
[nixpkgs-source]: https://github.com/NixOS/nixpkgs/tree/master/doc
[nixos-source]: https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual
[open documentation issues]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+label%3A%229.needs%3A+documentation%22
[open documentation pull requests]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+documentation%22%2C%226.topic%3A+documentation%22

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/nix/nixos-install-tools/default.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@ in
       nixos-install nixos-generate-config nixos-enter;
 
     # Required for --help.
```
Comment (roberth) [inline pkgs/tools/nix/nixos-install-tools/default.nix:23]:
```suggestion
```

This comment isn't accurate or helpful anymore.

**Review by roberth — COMMENTED** [review]
I like this. They're increasingly like normal packages, which I think is much nicer.
Can't test it now, but otherwise lgtm.

**Comment by roberth** [comment]
NixOS isn't quite happy though; see the ofborg error at `nixos/modules/misc/documentation.nix:349:36`

**Comment by doronbehar** [comment]
Thanks for the review and the support, I wasn't sure other will support this change. Pushed a change that hopefully will fix the ofborg errors. Will ping again when ofborg is completely green.

**Comment by doronbehar** [comment]
Pushed an additional commit that improves much the "contributing" documentation.

**Comment by doronbehar** [comment]
Tested also the build of the following attributes (ofborg listed and tested them earlier:

```
tests.testers.runNixOSTest-example
tests.nixos-functions.nixos-test
nixos-option
nixos-install-tools
nixos-rebuild
tests.testers.nixosTest-example
tests.trivial-builders.references
home-manager
```

**Comment by roberth** [comment]
> Pushed an additional commit that improves much the "contributing" documentation.

Ohh that's laudable but it probably conflicts with @infinisil's broader cleanup
 - https://github.com/NixOS/nixpkgs/pull/245243
 
Not sure what to do about that.

**Comment by doronbehar** [comment]
@roberth thanks for the notice. @infinisil's PR seems to be mostly about "how to contribute to docuemtation about *Nixpkgs'*", rather then "how to contribute to docuemtation about NixOS". Hence I'm fairly sure no conflicts will arise as the files he touches there do not intersect with the files changed here.

File: nixos/doc/manual/contributing-to-this-manual.chapter.md [hunk 2]
```diff
@@ -6,31 +6,104 @@ You can quickly check your edits with the following:
...
+For previewing edited files, you can just run `man --local-file path/to/file.8` and you will see it rendered.
+
+Being written in `mdoc`, these manpages use semantic markup. This file provides a guideline on where to apply which of the semantic elements of `mdoc`.
+
+### Command lines and arguments {#ssec-contributing-nixos-tools-cli-and-args}
+
+In any manpage, commands, flags and arguments to the *current* executable should be marked according to their semantics. Commands, flags and arguments passed to *other* executables should not be marked like this and should instead be considered as code examples and marked with `Ql`.
+
+- Use `Fl` to mark flag arguments, `Ar` for their arguments.
+- Repeating arguments should be marked by adding ellipses (`...`).
```
Comment (roberth) [inline nixos/doc/manual/contributing-to-this-manual.chapter.md:65]:
```suggestion
- Repeating arguments should be marked by adding an ellipsis (spelled with periods, `...`).
```

**Review by roberth — APPROVED** [review]
- Done a proofreading. Feel free to commit what you agree with.
- Tested in a VM
  - `man <tool>`
  - `man configuration.nix`
- Confirmed that we don't seem to have html pages for the nixos-* commands yet. Let's not do that in this pr.

:rocket: Can be merged when suggestions have been considered.

**Comment by doronbehar** [comment]
Thanks @roberth for the thorough review! Will give this a few hours today, and will merge afterwards.

**Pull request merged**